### PR TITLE
Staker nft

### PR DESCRIPTION
**Pull request by elclandestin0**
Staker NFT new metadata

Files added:
- `DateTime.sol`
    - The code is from another author, see @notice in file
    - Modified into a library
    - Modified solidity version

Files edited: 
- `PositionMetadata.sol`
    - Added new functions to generate svg on chain
- `Staker3.test.ts`
    - Filled 1st test to spit out tokenURIs of all token types after staking according to their respective token type

### CODE REVIEW

**Review by zomglings — DISMISSED** [review]
Approved, but I left some suggestions for improvement to make it easier on other reviewers. Nothing game breaking would be nice if you made those changes before you merged.

File: api/deploy/deploy.bash
Comment (elclandestin0) [inline api/deploy/deploy.bash:1]: What is this file? Says it's empty

Comment (elclandestin0) [inline api/deploy/deploy.bash:1]: Thought I made a comment. Says it's empty. Can we get rid of it?

Comment (mrk-hub) [reply api/deploy/deploy.bash:1]: File shouldn't be empty looks like issue from merge. Need to re-pull it from main

**Review by mrk-hub — APPROVED** [review]

**Pull request merged**